### PR TITLE
SC Config Path - Security review #7 fixes

### PR DESCRIPTION
**Pull request by nareddyt**
Various minor changes based on comments:
- `min_stream_report_interval_ms` should have a realistic minimum value enforced. Do this in our `ConfigParser` instead of proto config because we allow 0 value as default.
- `generated_header_prefix` is a header name, not value. Stricter validation here.
- `metric_cost_vector` should be stored as reference not array. It will never be null. Testing will be easier (test the values match, not the pointers).
- Rename `find_requirement` to match naming in the file. Seems we have some conflicting naming across filters, we can fix later.
- Clarify filter stats.
- Exception invalid access token case (already validated by protoc-gen-validate, so it should never be reached).

Signed-off-by: Teju Nareddy <nareddyt@google.com>

### CODE REVIEW

File: api/envoy/http/service_control/config.proto [hunk 1]
```diff
@@ -131,8 +131,7 @@ message FilterConfig {
 
   // The prefix added to generated headers
   string generated_header_prefix = 9 [(validate.rules).string = {
-    well_known_regex: HTTP_HEADER_VALUE,
-    strict: false,
+    well_known_regex: HTTP_HEADER_NAME,
```
Comment (TAOXUY) [inline api/envoy/http/service_control/config.proto:134]: Will header prefix `X-Endpoint-` be regarded as a valid header name?

Comment (nareddyt) [reply api/envoy/http/service_control/config.proto:134]: Yes. Our integration tests still pass, so it's definitely valid.

File: src/api_proxy/service_control/request_info.h [hunk 2]
```diff
@@ -122,7 +122,10 @@ struct CheckResponseInfo {
 struct QuotaRequestInfo : public OperationInfo {
   std::string method_name;
 
-  const std::vector<std::pair<std::string, int>>* metric_cost_vector;
+  const std::vector<std::pair<std::string, int>>& metric_cost_vector;
+
+  QuotaRequestInfo(const std::vector<std::pair<std::string, int>>& metric_costs)
```
Comment (qiwzhang) [inline src/api_proxy/service_control/request_info.h:127]: If you use &&, then you can use std::move() to swap it.

File: src/envoy/http/service_control/config_parser.h [hunk 3]
```diff
@@ -98,7 +112,8 @@ class FilterConfigParser {
       const {
     return config_;
   }
-  const RequirementContext* FindRequirement(absl::string_view operation) const {
+  const RequirementContext* find_requirement(
```
Comment (qiwzhang) [inline src/envoy/http/service_control/config_parser.h:115]: I want c++ code under "envoy" to follow envoy style. which is findRequirement

Comment (nareddyt) [reply src/envoy/http/service_control/config_parser.h:115]: Let me do this in another PR across the entire directory. 

This change makes it consistent with the rest of this file at least.

Comment (qiwzhang) [reply src/envoy/http/service_control/config_parser.h:115]: OK

**Pull request merged**